steel/src/host/mod.rs

```diff
@@ -103,3 +97,96 @@ where
         ))
     }
 }
+
+impl<H> EvmEnv<(), H> {
+    /// Returns a builder for this environment.
+    pub fn builder() -> EvmEnvBuilder<NoProvider, H> {
+        EvmEnvBuilder::default()
+    }
+}
+
+/// Builder to construct an [EvmEnv] on the host.
+#[derive(Clone, Debug)]
+pub struct EvmEnvBuilder<P, H> {
+    provider: P,
+    block: BlockNumberOrTag,
+    phantom: PhantomData<H>,
+}
+
+/// First stage of the [EvmEnvBuilder] without a specified [Provider].
+pub struct NoProvider {}
```

Comment:
Using the unit-like definition means that there is a single instantiation of the struct, which I think is more semantically aligned here.
```suggestion
pub struct NoProvider;
```